plugins/urls/cookiejar.py

```diff
@@ -0,0 +1,52 @@
+__author__ = 'Gareth Coles'
+
+from cookielib import LWPCookieJar
+
+
+class ChocolateCookieJar(LWPCookieJar):
+    mode = "save"  # Save, update, discard?
+
+    def set_mode(self, mode):
+        self.mode = mode
+
+    def set_cookie(self, cookie):
+        """Set a cookie, without checking whether or not it should be set."""
+        if self.mode == "discard":
+            return
+
+        c = self._cookies
+        self._cookies_lock.acquire()
+        try:
+            if self.mode in ("save", "session"):
+                if cookie.domain not in c:
+                    c[cookie.domain] = {}
+
+                c2 = c[cookie.domain]
+
+                if cookie.path not in c2:
+                    c2[cookie.path] = {}
+
+                c3 = c2[cookie.path]
+                c3[cookie.name] = cookie
+            else:
```

Comment:
`mode` is never checked for invalid values, so any incorrect value (typos are quite possible) will behave as if mode is "update". This should be checked somewhere (not necessarily in this class), and an error given if an invalid mode is given (especially considering how subtle this would be).